### PR TITLE
fixes bugs in post-RPC scheduling and cleanup

### PR DESCRIPTION
**Pull request by lukastruemper**
## Description
- RPC scheduling does not report loop nests changed, so post-cleanup is never triggered
- Transforms/schedulers do not respect already-scheduled loops from RPC, so the get double offloaded

## Related Issues

- Closes #

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [ ] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [ ] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [ ]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [ ] New / modified types have serialization / deserialization support
  *  [ ] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [ ] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [ ] Serializer has an option to emit the previous format

### Unit Tests
- [ ] New unit tests have been added for the changes.
- [ ] Existing unit tests pass locally.
- [ ] Edge cases and error paths are covered.

### Integration Tests
- [ ] New integration tests have been added (or existing ones updated).
- [ ] Integration tests pass locally.
- [ ] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [ ] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [ ] Test coverage has not decreased as a result of this change.
- [ ] Coverage report reviewed and acceptable.
- [ ] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

## Additional Context
Add any other context about the PR here (design decisions, trade-offs, follow-ups).

### CODE REVIEW

**Comment by NoraHagmeyer** [comment]
I fixed bugs related to double offloading a month ago and it worked for the tests I had. There are also unit tests ensuring that.

File: opt/src/passes/scheduler/cuda_scheduler.cpp [hunk 1]
```diff
@@ -23,6 +23,10 @@ SchedulerAction CUDAScheduler::find(
     structured_control_flow::StructuredLoop& loop,
     bool offload_unknown_sizes
 ) {
+    if (loop.schedule_type().category() != structured_control_flow::ScheduleTypeCategory::None) {
```
Comment (NoraHagmeyer) [inline opt/src/passes/scheduler/cuda_scheduler.cpp]: Add this check to loop_schedulining pass and return found_incompatible = true if a loop already has a schedule type. This way, it will work for all current and future schedulers.

File: sdfg/include/sdfg/passes/scheduler/loop_scheduler.h [hunk 2]
```diff
@@ -16,6 +16,7 @@ namespace scheduler {
 enum SchedulerAction {
     NEXT,
     CHILDREN,
+    SKIP,
```
Comment (NoraHagmeyer) [inline sdfg/include/sdfg/passes/scheduler/loop_scheduler.h]: So far, NEXT could also mean SKIP if nothing was done and then gone to the NEXT loop nest. If you introduce a difference, this logic should also be changed for consistency in all schedulers.

**Review by NoraHagmeyer — CHANGES_REQUESTED** [review]
Thanks for fixing that! I though I already did but seem to have only done so for multi-dimensional offloading.

Ibelieve the loop_scheduling_pass is the better location for the check though.
